### ISSUE TITLE
NACs Phase Consistency

### ISSUE THREAD

**Issue by naythanyeo** [rhyan10/X-MACE#6]
Hi, I was using X-MACE and trying to train with NACs but noticed some possible inconsistencies with how the phase invariance is being handled in the current implementation. 

The current loss and trainer evaluates the NAC errors with a phase invariant loss by taking the minimum difference between the predicted and actual (up to a sign)
neg = torch.sum(torch.square(ref["nacs"] - pred["nacs"]), dim=-1)
pos = torch.sum(torch.square(ref["nacs"] + pred["nacs"]), dim=-1)
err2 = torch.minimum(pos, neg)

To the best of my understanding, this is done typically because NACs are relative to the sign of the wavefunction, which can take on two equivalent values. 
However, the current implementation takes the minimum of each atom's NAC values, which allows the NAC sign to be chosen independently for different atoms
This also makes it such that the relative signs of the NAC pairs are treated independently. 
As such the current loss and test metrics for NACs are overly permissive to the sign flips. 

Two constraints I think should be added
1) Within each molecular geometry in the batch, atoms should be constrained to follow the same sign convention
2) To take the minimum NAC error between the different states, you could enumerate all allowed electronic state phase assignments and take the minimum of the various combinations, so the difference states are not treated independently. https://github.com/schnarc/SchNarc/blob/master/src/schnarc/nn.py has a version that implements the loss this way 

If not the NAC errors will be artificially lowered when you report them

**Comment by naythanyeo** [comment]
[nac_utils.py](https://github.com/user-attachments/files/32153454/nac_utils.py)

I've written some helpers that return a phase aligned residue error
These can be called later in loss / train metric with 

nac_residue = align_batch_nacs(
       pred=output["nacs"],
       ref=batch["nacs"],
       ptr=batch.ptr,
       num_states=batch["energy"].shape[-1]
)

MAE = torch.mean(nac_residue.abs()) 
or
MSE = torch.mean(nac_residue.square())
